### PR TITLE
Update info

### PR DESCRIPTION
**Pull request by vieux**
fix https://github.com/docker/swarm/issues/1671

ping @dongluochen

### CODE REVIEW

**Comment by MHBauer** [comment]
wow. I just realized that swarm could be running on windows and absolutely nothing would change. 

blows my mind :open_mouth:

**Comment by nishanttotla** [comment]
It seems like you're setting a default value for `KernelVersion` (`"<unknown>"`) for the `/info` endpoint, but for the `/version` endpoint there's no such default. Can we make that consistent?

Apart from that, LGTM.

**Comment by vieux** [comment]
@nishanttotla thanks, I updated

**Comment by nishanttotla** [comment]
Great, LGTM :)

**Comment by dongluochen** [comment]
Is stable docker API version not showing container categories like 'Running', 'Stopped'? In that case we may need to skip the test for stable.

**Comment by nishanttotla** [comment]
@dongluochen I think `engine-api` was going to be vendored in at the end of the week (as per @calavera). Should that be enough?

**Comment by vieux** [comment]
@nishanttotla stable means engine 1.9 and this feature is only present in 1.10

**Comment by vieux** [comment]
thanks @dongluochen I'll update

**Comment by vieux** [comment]
test failure are unrelated, they will probably be fixed by https://github.com/docker/swarm/pull/1690

@dongluochen can you please review again ?

**Comment by dongluochen** [comment]
LGTM.

File: api/handlers.go [hunk 1]
```diff
@@ -29,23 +29,45 @@ const APIVERSION = "1.21"
...
 		NoProxy:           os.Getenv("no_proxy"),
 		SystemTime:        time.Now().Format(time.RFC3339Nano),
+		ExperimentalBuild: experimental.ENABLED,
+	}
+
+	if kernelVersion, err := kernel.GetKernelVersion(); err == nil {
+		info.KernelVersion = kernelVersion.String()
+	}
+
+	for _, c := range c.cluster.Containers() {
```
Comment (abronan) [inline api/handlers.go]: @vieux I don't think this is the most efficient way to do it, we are looping across all the containers in the cluster, if we have many tens of thousands of containers, this is a bit overkill (especially knowing that `Containers()` also loops through all the engine's container entries first). `info` provides the number of `Paused` and `Running` containers now thanks to changes in the Engine API, might be better to use those values instead I think.

Knowing that the rate of creation of containers is pretty stable after going through the scheduling process, calculating this from the cached values on `info` for every engine will be more efficient and still provide with meaningful values (unless you run `docker info` every second :smile:).

Comment (vieux) [inline api/handlers.go]: ok if you don't mind, merge this one and I'll start working on this on another PR

Comment (abronan) [inline api/handlers.go]: Sounds perfect :) thanks @vieux!

**Pull request merged**